src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java

```diff
@@ -142,7 +143,11 @@ public Value call() throws Exception {
                 } else {
                     BitDocIdSet.Builder builder = new BitDocIdSet.Builder(context.reader().maxDoc());
                     if (docIdSet != null && docIdSet != DocIdSet.EMPTY) {
-                        builder.or(docIdSet.iterator());
+                        DocIdSetIterator iterator = docIdSet.iterator();
+                        // some filters (QueryWrapperFilter) return not null or DocIdSet.EMPTY if there no matching docs
+                        if (iterator != null) {
+                            builder.or(iterator);
+                        }
```

Comment:
Note: I was wrongly assuming that DocIdSet.EMPTY returns a null iterator, which is not the case.